os/kernel/binary_manager/binary_manager_binfile.c

```diff
@@ -178,7 +179,7 @@ int binary_manager_create_entry(int requester_pid, char *bin_name, int version)
 		}
 	} else {
 		/* If it it not registered, register it */
-		ret = binary_manager_register_ubin(bin_name);
+		ret = binary_manager_register_ubin(bin_name, SCHED_PRIORITY_MIN);
```

Comment:
Why do you try to register with lowest priority?
Isn't it  possible to register with real priority?

Reply:
Yes. we don't know a priority of new binary to be downloaded.